components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java

```diff
@@ -230,11 +230,11 @@ public void handleEvent(Event event) throws IdentityEventException {
                 AuthorizedAPI authorizedAPI = OAuthComponentServiceHolder.getInstance()
                         .getAuthorizedAPIManagementService()
                         .getAuthorizedAPI(appId, apiId, tenantDomain);
-                List<String> removedScopes = new ArrayList<>();
-                removedScopes.addAll(authorizedAPI.getScopes().stream()
-                                .map(Scope::getName).filter(scope ->
-                        !removedScopes.contains(scope)).collect(Collectors.toList()));
-
+                if (authorizedAPI.getScopes() == null) {
+                    return;
+                }
+                List<String> removedScopes = authorizedAPI.getScopes().stream()
```

Comment:
Will this handle duplicate scopes, don't we need to check removedScopes.contains(scope) before adding or we can make it as `set`

Reply:
there can't be any duplicated scopes coming in the authorized scopes list